app/src/main/scala/org/alephium/explorer/service/TokenSupplyService.scala

```diff
@@ -223,12 +222,15 @@ case object TokenSupplyService extends TokenSupplyService with StrictLogging {
       DBIOAction.sequence(
         groupSetting.groupIndexes.map {
           case (from, to) =>
-            BlockHeaderSchema.table
-              .filter(header => header.chainFrom === from && header.chainTo === to)
-              .sortBy(_.timestamp.desc)
-              .map(_.timestamp)
-              .result
-              .headOption
+            sql"""
+              SELECT block_timestamp
+              FROM block_headers
+              WHERE main_chain = true
+              AND chain_from = $from
+              AND chain_to = $to
+              ORDER BY block_timestamp DESC
+              LIMIT 1
```

Comment:
The typed query being replaced is not querying for `main_chain = true`. Is the `main_chain` check in SQL query something that was missing from the typed query?

Reply:
Good point, I think it actually doesn't change much with or without it, token supply is computed daily at midnight, the latest  block is more like a time reference to see when we change day. 
As it's the same it's maybe cleaner to have the `main_chain = true`?